apps/web/app/dashboard/bookmarks/page.tsx

```diff
@@ -1,20 +1,40 @@
-import React from "react";
-import Bookmarks from "@/components/dashboard/bookmarks/Bookmarks";
-import TopNav from "@/components/dashboard/bookmarks/TopNav";
+"use client";
+
+import { Suspense, useRef } from "react";
+import BookmarksGrid from "@/components/dashboard/bookmarks/BookmarksGrid";
+import { SearchInput } from "@/components/dashboard/search/SearchInput";
+import { FullPageSpinner } from "@/components/ui/full-page-spinner";
 import { Separator } from "@/components/ui/separator";
+import { useBookmarkSearch } from "@/lib/hooks/bookmark-search";
+
+function BookmarksComp() {
+  const { data } = useBookmarkSearch();
```

Comment:
Relying on the data returned from the search endpoint in the homepage is not a great idea. The search endpoint depends on the data being fetched from meillisearch. Also, if mellisearch is disabled, this endpoint won't return any data.

If you really want to get rid of the redirection to the search page, I'd make the distinction between empty query string (in which case you use the `Bookmarks` component) and a non-empty query string, in which case you use the search endpoint and render the `BookmarkGrid`.